### DISCUSSION TITLE
Connect constructor not generated

### DISCUSSION THREAD

**Discussion by frankvollebregt** [simolus3/drift#1534] [Q&A] [answered]
I cloned my code on my new machine, after which this error popped up. I didn't yet migrate to the new `Drift` name, so that's the first thing I tried, to no avail.

In my own `mydatabase.dart` file, I have the constructor that makes a call to the `connect` constructor of the `GeneratedDatabase`:
```dart
  MyDatabase.connect(DatabaseConnection connection) : super.connect(connection);
```
But the call to super is highlighted, saying the following:
```
The class '_$MyDatabase' doesn't have a constructor named 'connect'.
Try defining a constructor named 'connect' in '_$MyDatabase', or invoking a different constructor.
```
When I look into the generated `MyDatabase` file, I see it extends `GeneratedDatabase`:
```dart
abstract class _$MyDatabase extends GeneratedDatabase {
```
Though only with a single constructor like this:
```dart
_$MyDatabase(QueryExecutor e) : super(SqlTypeSystem.defaultInstance, e);
```
Now on my old machine (and before the `Drift` migration), the generated file contains both constructors, like so:
```dart
_$MyDatabase(QueryExecutor e) : super(SqlTypeSystem.defaultInstance, e);
_$MyDatabase.connect(DatabaseConnection c) : super.connect(c);
```
Did something change, or did I accidentally change something myself? Thanks!

As a note, if I manually add the second constructor to my generated file on the new machine, everything works as expected.

Versions:
- build_runner: ^2.1.4
- drift_dev: ^1.0.2

- drift: ^1.0.1
- sqlite3_flutter_libs: ^0.5.1
- sqlparser: ^0.18.0
- path_provider: ^2.0.6
- path: ^1.8.0

**Comment by kuhnroyal** [comment, marked as answer]
You need to activate the required builder option: https://drift.simonbinder.eu/docs/advanced-features/builder_options/

**Reply by simolus3** [reply]
This is probably it. You can copy over the `build.yaml` file without many changes. Note that you may have to `moor_generator` builders with the corresponding `drift_dev` as described [here](https://drift.simonbinder.eu/name/#build-configuration) (although the automated migration tool should have done that for you).

**Reply by frankvollebregt** [reply]
Thank you! That was indeed the issue. I haven't had to use a `build.yaml` before at all, but now I added the example with the extra key to generate the connect constructor and it works as expected.

```yaml
targets:
  $default:
    builders:
      drift_dev:
        options:
          compact_query_methods: true
          generate_connect_constructor: true
```
